android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -90,9 +83,17 @@ class RNIapModule(
             return
         }
 
+        if (billingClient?.isReady == true) {
+            Log.i(
+                TAG,
+                "Already initialized, you should only call initConnection() once when your app starts"
+            )
+            promise.safeResolve(true)
+            return
+        }
         billingClient = builder.setListener(this).build()
```

Comment:
Would there be any chance on `billingClient` isn't `null` but `isReady` is `false`?
If so, do you think the garbage collector here would make such a problem?

Reply:
I changed it so that it the instance gets disposed on `endConnection`